six/modules/c++/six.sicd/include/six/sicd/Utilities.h

```diff
@@ -358,6 +383,99 @@ class Utilities
      * \return mock ComplexData object
      */
     static std::auto_ptr<ComplexData> createFakeComplexData();
+
+    /*
+     * Given a reference to a loaded NITFReadControl, this function
+     * parses the SICD's DES and returns a NoiseMesh if present.
+     * \param reader A NITFReadControl loaded with the desired SICD
+     * \return Noise Mesh associated with the SICD NITF
+     * \throws except::Exception if the provided reader is not a SICD
+     *
+     */
+    static std::auto_ptr<NoiseMesh> getNoiseMesh(NITFReadControl& reader);
+
+    /*
+     * Given a reference to a loaded NITFReadControl, this function
+     * parses the SICD's DES and returns a a vector of Mesh objects.
+     * \param reader A NITFReadControl loaded with the desired SICD
+     * \param[out] outputRowColToSlantRow Projection polynomial
```

Comment:
Need to describe units here.  If you want to defer all the detailed unit definition to one class/spot and refer to that everywhere, that seems reasonable too.